packages/autorest.typescript/src/utils/nameConstructors.ts

```diff
@@ -0,0 +1,35 @@
+export type ModuleName =
+  | string
+  | {
+      esModulesName: string;
+      cjsName: string;
+    };
+
+/**
+ * This is a helper function that gets the right import module depending on the type of
+ * library being generated.
+ *
+ * @param name The name of the module to import - this can be a string representing the base name of the module or an object with the cjsName and esModulesName properties. *
+ * @param moduleKind The type of module being generated. This can be either "cjs" or "esm".
+ *
+ * @example
+ *
+ * ```ts
+ * getImportModuleName("myModule", "cjs") // returns "myModule"
+ * getImportModuleName("myModule", "esm") // returns "myModule.js"
```

Comment:
q: Does this handle `getImportModuleName("myModule.js", "esm")`? Or `getImportModuleName("myModule.js", "cjs")`?

Reply:
Good Q! Right now it does not. It expects a baseName as a parameter which is what we provide in all usages. If we need to enhance this we can assuming there's a good usecase. Otherwise, we can wait for the need to arise since this is opt-in